File: C#/Transactions/How_To_Use_Token.cs
/*
* BluePay C#.NET Sample code.
*
* Charges a customer $3.00 using the payment and customer information from a previous transaction.
* If using TEST mode, odd dollar amounts will return
* an approval and even dollar amounts will return a decline.
*/

using System;
using BluePayLibrary;

namespace Transactions
{
    class HowToUseToken
    {
        public static void Main()
        {
            string accountID = "Merchant's Account ID Here";
            string secretKey = "Merchant's Secret Key Here";
            string mode = "TEST";
            string token = "Transaction ID Here"; 
        
            BluePay payment = new BluePay
            (
                accountID,
                secretKey,
                mode
            );

            //payment.SetCustomerInformation
            //(
            //    storedIndicator: "F",
            //    storedType:"C",
            //    storedId:"TESTID765456"
            //);

            // Set Sale Amount: $3.00.
            payment.Sale
            (
                amount: "3.00",
                masterID: token // Transaction ID of a previous sale
            );

            // Makes the API Request with BluePay
            payment.Process();

            // If transaction was successful reads the responses from BluePay
            if (payment.IsSuccessfulTransaction())
            {
                Console.WriteLine("Transaction Status: " + payment.GetStatus());
                Console.WriteLine("Transaction ID: " + payment.GetTransID());
                Console.WriteLine("Transaction Message: " + payment.GetMessage());
                Console.WriteLine("AVS Response: " + payment.GetAVS());
                Console.WriteLine("CVV2 Response: " + payment.GetCVV2());
                Console.WriteLine("Masked Payment Account: " + payment.GetMaskedPaymentAccount());
                Console.WriteLine("Card Type: " + payment.GetCardType());
                Console.WriteLine("Authorization Code: " + payment.GetAuthCode());
                //Console.WriteLine("Stored ID: " + payment.GetStoredId());
            }
            else
            {
                Console.WriteLine("Error: " + payment.GetMessage());
            }
        }
    }
}

File: PHP/Transactions/How_To_Use_Token.php
<?php
/**
* BluePay PHP Sample Code
*
* Charges a customer $3.00 using the payment information from a previous transaction. 
* If using TEST mode, odd dollar amounts will return
* an approval and even dollar amounts will return a decline.
*/

include('../BluePay.php');

$accountID = "Merchant's Account ID Here";
$secretKey = "Merchant's Secret Key Here";
$mode = "TEST";
$token = "Transaction ID here"; 

$payment = new BluePay(
    $accountID,
    $secretKey,
    $mode
);

$payment->sale(array(
	'amount' => '3.00', 
	'masterID' => $token 
));

$payment->process();

if($payment->isSuccessfulResponse()){
// Read response from BluePay
echo 
	'Transaction Status: '. $payment->getStatus() . "\n" .
	'Transaction Message: '. $payment->getMessage() . "\n" .
	'Transaction ID: '. $payment->getTransID() . "\n" .
	'AVS Response: ' . $payment->getAVSResponse() . "\n" .
	'CVS Response: ' . $payment->getCVV2Response() . "\n" .
	'Masked Account: ' . $payment->getMaskedAccount() . "\n" .
	'Card Type: ' . $payment->getCardType() . "\n" .
	'Authorization Code: ' . $payment->getAuthCode() . "\n";
} else{
    echo $payment->getMessage() . "\n";
}
?>
